shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -517,6 +515,14 @@ - (void)updateDisplayConfig {
   return _bridge;
 }
 
+- (nonnull NSString*)executableName {
+  NSArray<NSString*>* arguments = [[NSProcessInfo processInfo] arguments];
+  if ([arguments count] < 1) {
+    return @"Flutter";
+  }
+  return [arguments objectAtIndex:0];
```

Comment:
Optional: you can make this whole method a one-liner with the (ObjC-style-guide-approved) Elvis operator:

```
return [[[NSProcessInfo processInfo] arguments] firstObject] ?: @"Flutter";
```

Reply:
Done!